### PR TITLE
Preliminary FFS-based MxN data transfer for global arrays

### PR DESCRIPTION
**Pull request by eisenhauer**
works with heatTransfer.

### CODE REVIEW

**Comment by pnorbert** [comment]
Same typename trick is required on line 61
/home/adios/work/worktree/ADIOS2/release/source/adios2/engine/sst/SstReader.cpp: In lambda function:
/home/adios/work/worktree/ADIOS2/release/source/adios2/engine/sst/SstReader.cpp:61:31: error: ‘Reader’ was not declared in this scope
         SstReader::SstReader *Reader =

**Comment by eisenhauer** [comment]
Also added to line 61.

File: source/adios2/engine/sst/SstReader.cpp [hunk 1]
```diff
@@ -58,8 +58,8 @@ SstReader::SstReader(IO &io, const std::string &name, const Mode mode,
         std::vector<size_t> VecStart;
         std::vector<size_t> VecCount;
         std::string Type(type);
-        SstReader::SstReader *Reader =
-            reinterpret_cast<SstReader::SstReader *>(reader);
+        typename SstReader::SstReader *Reader =
+            reinterpret_cast<typename SstReader::SstReader *>(reader);
```
Comment (chuckatkins) [inline source/adios2/engine/sst/SstReader.cpp]: Why would this need `typename` when there are no template arguments.

Comment (pnorbert) [reply source/adios2/engine/sst/SstReader.cpp]: My Ubuntu 16.04, g++ 5.4.0 did not like that line. ```typename``` fixed it. Is there a better fix?

**Comment by chuckatkins** [comment]
Most recent push was a rebase on current release

**Pull request merged**